packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -744,37 +756,51 @@ class ServiceExtensionManager {
     final expectedValueType =
         extensions.serviceExtensionsAllowlist[name].values.first.runtimeType;
 
-    try {
-      final response = await _service.callServiceExtension(
-        name,
-        isolateId: _isolateManager.selectedIsolate.id,
-      );
-      switch (expectedValueType) {
-        case bool:
-          final bool enabled =
-              response.json['enabled'] == 'true' ? true : false;
-          await _maybeRestoreExtension(name, enabled);
-          return;
-        case String:
-          final String value = response.json['value'];
-          await _maybeRestoreExtension(name, value);
-          return;
-        case int:
-        case double:
-          final num value = num.parse(
-              response.json[name.substring(name.lastIndexOf('.') + 1)]);
-          await _maybeRestoreExtension(name, value);
-          return;
-        default:
-          return;
+    final restore = () async {
```

Comment:
can't this can just be
`void restore() async {`
?
I don't see anywhere where you depend on equality for the closure.

Reply:
you're right. fixed